chessboard/gallen/chessboard.go

```diff
@@ -0,0 +1,40 @@
+package chessboard
+
+const rankSize = 8
+
+type Rank [rankSize]bool
+
+type Chessboard map[string]Rank
+
+func CountInRank(cb Chessboard, rank string) (count int) {
+    for _, occupied := range cb[rank] {
```

Comment:
I think that there should be a check to see if `rank` is a key in `cb`. It would be something like `r, ok := cb[rank]`. `ok` is boolean of if rank is key in `cb`.

Reply:
Actually, you don't need that.  From https://go.dev/doc/effective_go#maps:
```
An attempt to fetch a map value with a key that is not present in the map will return the zero value for the type of the entries in the map. 
```
In this case I believe it will return `nil`, and the range of nil is nothing (or nil) so the loop doesn't execute at all.

Reply:
It does work indeed. Nice.